packages/next/src/server-build.ts

```diff
@@ -1111,23 +1187,29 @@ export async function serverBuild({
     isEmptyAllowQueryForPrendered,
   });
 
+  const promises: Promise<void>[] = [];
+
   Object.keys(prerenderManifest.staticRoutes).forEach(route =>
-    prerenderRoute(route, {})
+    promises.push(prerenderRoute(route, {}))
   );
   Object.keys(prerenderManifest.fallbackRoutes).forEach(route =>
-    prerenderRoute(route, { isFallback: true })
+    promises.push(prerenderRoute(route, { isFallback: true }))
   );
   Object.keys(prerenderManifest.blockingFallbackRoutes).forEach(route =>
-    prerenderRoute(route, { isBlocking: true })
+    promises.push(prerenderRoute(route, { isBlocking: true }))
   );
 
   if (static404Page && canUsePreviewMode) {
     omittedPrerenderRoutes.forEach(route => {
-      prerenderRoute(route, { isOmitted: true });
+      promises.push(prerenderRoute(route, { isOmitted: true }));
     });
   }
 
+  // Wait for all prerenders to finish.
+  await Promise.all(promises);
```

Comment:
This can be a race condition where if it errors the builder doesn't tolerate the uncaught exception we need to either use Promise.all on each of these with `.map` or we need to add `.catch` to each individual promise and handle with `process.exit(1)` and proper log